packages/composite-checkout-wpcom/src/types/wpcom-store-state.ts

```diff
@@ -0,0 +1,271 @@
+/*
+ * All child components in composite checkout are controlled -- they accept
+ * data from their parents and evaluate callbacks when edited, rather than
+ * managing their own state. Hooks providing this data in turn need some extra
+ * data on each field: specifically whether it has been edited by the user
+ * or passed validation. We wrap this extra data into an object type.
+ */
+interface ManagedValue< T > {
+	value: T;
+	isTouched: boolean; // Has value been edited by the user?
+	errors: string[]; // Has value passed validation?
+}
+
+export function isValid< T >( arg: ManagedValue< T > ): boolean {
+	return arg.errors?.length <= 0;
+}
+
+function initialManagedValue< T >( value: T ): ManagedValue< T > {
+	return {
+		value: value,
+		isTouched: false,
+		errors: [ '' ],
+	};
+}
+
+function touchIfDifferent< T >( newValue: T, oldData: ManagedValue< T > ): ManagedValue< T > {
+	return newValue === oldData.value ? oldData : { ...oldData, value: newValue, isTouched: true };
+}
+
+function setErrors< T >(
+	errors: string[] | undefined,
+	oldData: ManagedValue< T >
+): ManagedValue< T > {
+	return undefined === errors ? { ...oldData, errors: [] } : { ...oldData, errors };
+}
+
+/*
+ * The wpcom store hook stores an object with all the contact info
+ * which is used to share state across fields where appropriate.
+ * Each value keeps track of whether it has been edited and validated.
+ */
+export type ManagedContactDetails = {
+	firstName: ManagedValue< string >;
+	lastName: ManagedValue< string >;
+	organization: ManagedValue< string >;
+	email: ManagedValue< string >;
+	alternateEmail: ManagedValue< string >;
+	phone: ManagedValue< string >;
```

Comment:
Since the names of these fields have changed (eg: `phoneNumber` to `phone`, `country` to `countryCode`), they will break anywhere that currently uses `getContactInfo()`, notably the payment method getters defined in `createPaymentMethods()`. We should find those places and change them too.

Reply:
I think I fixed all of these.